src/App.tsx

```diff
@@ -15,48 +17,104 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+const SORT_FIELD_LENGTH = 'length';
+const SORT_FIELD_ALPH = 'alphabetically';
+
+interface SortParams {
+  sortField: string;
+  isReversed: boolean;
+}
+
+function getPrreparedGoods(
+  goods: string[],
+  { sortField, isReversed }: SortParams,
+) {
+  const prepearedGoods = [...goods];
+
+  if (sortField) {
+    prepearedGoods.sort((a, b) => {
+      switch (sortField) {
+        case SORT_FIELD_LENGTH:
+          return a.length - b.length;
+
+        case SORT_FIELD_ALPH:
+          return a.localeCompare(b);
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  return isReversed ? prepearedGoods.reverse() : prepearedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+
+  const visibleGoods = getPrreparedGoods(goodsFromServer, {
+    sortField,
+    isReversed,
+  });
+
+  const causeButton = isReversed === true || sortField;
+
+  const resetButton = () => {
+    setSortField('');
+    setIsReversed(false);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={cn('button', 'is-info', {
+            'is-light': sortField !== SORT_FIELD_ALPH,
+          })}
+          onClick={() => setSortField(SORT_FIELD_ALPH)}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={cn('button', 'is-success', {
+            'is-light': sortField !== SORT_FIELD_LENGTH,
+          })}
+          onClick={() => setSortField(SORT_FIELD_LENGTH)}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={cn('button', 'is-warning', {
+            'is-light': isReversed === false,
+          })}
+          onClick={() => setIsReversed(prev => !prev)}
```

Comment:
Consider will be better if you create a helper function for reverse and use it here